### ISSUE TITLE
Convention to specify a link for bibliography

### ISSUE THREAD

**Issue by danmackinlay** [cormacrelf/citeproc-rs#59]
As per the suggestion there, this issue is [cloned from a citeproc-js issue](https://github.com/Juris-M/citeproc-js/issues/128):

> Currently AFAICT [CSL is in limbo regardingspecification for hyperlinks in bibliographies](https://github.com/citation-style-language/schema/issues/145) and so CSL implementers must invent their own convention for specifying hyperlinks. Is this something we are interested in? Do we need some way of specifying that part or all of a bibliographical entry (such as the whole entry, or the title, or the author name/date or something) should be a hyperlink (either to the `URL` field or the `DOI` link)? Currently it seems there is the `wrap_url_and_doi` process which can be ugly, although of course one can make it look nicer with CSS for HTML output. if one has control of the CSS. There are also [macro hacks](https://github.com/jgm/pandoc-citeproc/issues/318) for LaTeX output; but it seems like some kind of unified approach might be in order?

**Comment by cormacrelf** [comment]
Hey guess what, citeproc-rs already has its own version of `wrap_url_and_doi`! It has a number of shortfalls, and it frankly shouldn't be enabled by default given the state of it:

-  I haven't put proper escaping on it so weird URLs will break
- haven't implemented it for RTF
- there isn't an API for supplying flags to the processor at runtime (the way citeproc-js can be configured via JavaScript)
- won't convert non-link DOIs into proper URLs
- won't recognise inputs that aren't URLs and refuse to wrap them in a link.

So it's not done in any sense of the word. But if you put a very simple link in there it will get an anchor tag around it in HTML, and the affixes work correctly so \<[https://google.com](https://google.com)\> renders correctly.

For reference, the code is roughly in [these](https://github.com/cormacrelf/citeproc-rs/blob/67441edeb4f1a204f36dc25514723e9a3725cd98/crates/io/src/output/markup.rs#L192), [three](https://github.com/cormacrelf/citeproc-rs/blob/67441edeb4f1a204f36dc25514723e9a3725cd98/crates/io/src/output/markup/html.rs#L129-L139), [places](https://github.com/cormacrelf/citeproc-rs/blob/67441edeb4f1a204f36dc25514723e9a3725cd98/crates/proc/src/renderer.rs#L398-L401)

The [definition of `InlineElement::Anchor` with children being nodes rather than a string](https://github.com/cormacrelf/citeproc-rs/blob/67441edeb4f1a204f36dc25514723e9a3725cd98/crates/io/src/output/markup.rs#L60) means it's very easy to support wrapping other output like affixes etc, but I would prefer to avoid adding a feature like this:

```xml
<group urlwrap="DOI">
  <text value="doi:" />
  <text variable="DOI" />
</group>
```

(Output is [doi:12345](https://example.com) as opposed to doi:[12345](https://example.com).)

Essentially because groups being implicit conditionals means you need to change the definition of an intermediate group, and then add hyperlinks when flatting those groups later, which is not ideal. Keeping that inside the `<text>` element keeps the hyperlinking from being another 'deferred' attribute:

```xml
<text url="DOI" include-affixes="true" prefix="doi:" />
```

Basically, `text url` could be another variant of `text variable/value/macro` etc. I would be cool with that.
